### ISSUE TITLE
Reminder due date keeps changing to next day, or is cleared out.

### ISSUE THREAD

**Issue by josh-nz** [spacecowboy/NotePad#284]
Things used to work great, then a while back now I noticed that my reminder notifications were not always triggering. When I view the reminder again, the due date would be due date + 1. If I then waited until the following day, again the notification wouldn't fire. Viewing the reminder would display the due date as due date + 1 again. This repeats forever.

It doesn't happen on all reminders. Some fire correctly, others have the above behaviour. I tried uninstalling the app and reinstalling, same issue. I've also disable Google Tasks sync, and still have the same issue.

As such, I cannot rely upon this app. Which is a shame, it's the best reminder app for me.

I'm sure I posted a bug report about this somewhere when I first noticed this issue. I never got any reply, but now I can't find it. :) So here's a new one.

(Also, it may have been that some of the reminders that didn't fire notifications simply had cleared out the due date, rather than set it to due date + 1, so I think there are two, possibly related, issues here. It's been a while since I've really used the app on account of this unreliability).

I'm happy to help test beta versions with logging or whatever else is needed to get more info on these issues.

Thanks.

**Comment by spacecowboy** [comment]
Thanks for the report.
If I interpret you correctly, you setup a reminder and configures it to repeat every day. While reminders without repetition might simple never fire.

I'll investigate and see if I can reproduce it.

**Comment by josh-nz** [comment]
No, you're interpretation is not correct. Here's a set of steps I take:

1. Touch + to create a new note
2. Add some text to the 'note' field
3. Touch 'add a reminder' link
4. Touch the date and select a date
5. Touch the time and select a time
6. Touch < Notes link at top left of screen to save note

I do NOT set more than 1 reminder per note.
I do NOT schedule the reminder for one or more days. (ie, mon, tues, wed, thu, fri, sat, sun texts do NOT have a blue line under any of them).

When the date and time as set in steps 4 and 5 above occurs, one of 2 things will happen:

Either:
a. The notification will trigger as expected, or
b. The notification will NOT trigger at all

In the case of b where there is no notification, if I then view the note I see one of 2 things:

Either:
aa. The reminder date and time set in steps 4 and 5 above now reads (date + 1 day), time, or
bb. The reminder date and time set in steps 4 and 5 above has been removed completely, as though I had never set one.

I hope this is clearer.

Note: When I uninstalled the app and reinstalled it, my existing notes came back. I was not using Google Tasks sync. This suggests that uninstalling the app still leaves some data behind on my device. I would like to totally uninstall the app and do a clean install to see if this fixes the issue - perhaps something is cached that is causing this incorrect reminder behaviour. Where can I find any remaining files to remove after uninstalling the app so that a reinstall will be clean?

Thanks for your time.

**Comment by spacecowboy** [comment]
Thanks for the clarification. I'll try to reproduce it with the new steps.

If you uninstall the app, everything should be deleted. But if you've used sync in the past, reinstalling fron the Play store can restore your last used settings which will trigger a sync directly. Same if you used SD-card sync.

To force a clean install, just clear the app's data from settings:

Settings->Apps

![2015-01-12 10 06 55](https://cloud.githubusercontent.com/assets/223655/5701430/468bcdb2-9a4b-11e4-8123-f87ee40a1ba8.png)

Select the app from the list, and then select "Clear data" and "Clear cache" in here:
![2015-01-12 10 07 05](https://cloud.githubusercontent.com/assets/223655/5701435/604fac5a-9a4b-11e4-8ac4-a143b0b9aac4.png)

**Comment by josh-nz** [comment]
Just a quick update. I cleared the app's data from settings, uninstalled the app and reinstalled. I've had the issue where the reminder keeps incrementing by a day, each day, so it therefore never goes off (outcome `aa` in my earlier message) come back on one todo.

So even on a 'clean' install, I get the issue.

**Comment by spacecowboy** [comment]
Thanks for the added details. I'll keep trying to reproduce it.

**Comment by licaon-kter** [comment]
I now observe this in 6beta5 (I only use notes not tasks; sync on SD), basically the auto added reminder is updated on app (or note) opening, while manually edited ones do not. This original issue talks about a day, now this happens to an hour since #298 got implemented.

This is some Schrodiger cat in the box thing.

Steps:
* Write or edit a note
* Add reminder (do not edit it) it will be say at 11:23 
* Add another note
* Set reminder
* Edit reminder for 11:30
* Close the app

Now wait 5 minutes and open the app:
* The first note when you open it will have the reminder at 11:28
* The second one wil be still at 11:30

I'll keep tabs if the reminders at least triggers on the first time I set up, and I'll add this info here.

I guess ths will be fixed with #306 since now the reminder date/time is only kept in app data and not exported to the SD card.
